source/scpp/src/crypto/ByteSliceHasher.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "ByteSliceHasher.h"
+#include <sodium.h>
+
+namespace stellar
+{
+namespace shortHash
+{
+static unsigned char sKey[crypto_shorthash_KEYBYTES];
+void
+initialize()
+{
+    crypto_shorthash_keygen(sKey);
```

Comment:
This has to be called so you might want to stick a `shared static this` in the bindings